tasks/github_tasks.py

```diff
@@ -110,6 +113,56 @@ def trigger_macos(
         raise Exit(message=f"Macos {workflow_type} workflow {conclusion}", code=1)
 
 
+def _update_windows_runner_version(new_version=None, buildenv_ref="master"):
+    if new_version is None:
+        raise Exit(message="Buildenv workflow need the 'new_version' field value to be not None")
+
+    run = trigger_buildenv_workflow(
+        workflow_name="runner-bump.yml",
+        github_action_ref=buildenv_ref,
+        new_version=new_version,
+    )
+    # We are only waiting 0.5min between each status check because buildenv is much faster than macOS builds
+    workflow_conclusion, workflow_url = follow_workflow_run(run, "DataDog/buildenv", 0.5)
+
+    if workflow_conclusion == "failure":
+        print_failed_jobs_logs(run)
+    else:
+        raise Exit(message="Workflow failed, see errors below.")
+
+    print_workflow_conclusion(workflow_conclusion, workflow_url)
+
+    download_with_retry(download_artifacts, run, ".", 3, 5, "DataDog/buildenv")
+
+    with open("PR_URL_ARTIFACT") as f:
+        PR_URL = f.read().strip()
+
+    if not PR_URL:
+        raise Exit(message="Failed to fetch artifact from the workflow. (Empty artifact)")
+
+    message = f":robobits: A new windows-runner bump PR to {new_version} has been generated. Please take a look :frog-review:\n:pr: {PR_URL} :ty:"
+
+    send_slack_message("ci-infra-support", message)
+    return workflow_conclusion
+
+
+@task
+def update_windows_runner_version(
+    ctx,
+    new_version=None,
+    buildenv_ref="master",
+):
+    """
+    Trigger a workflow on the buildenv repository to bump windows gitlab runner
+    """
+    if new_version is None:
+        new_version = str(current_version(ctx, "7"))
+
+    conclusion = _update_windows_runner_version(new_version, buildenv_ref)
+    if conclusion != "success":
+        raise Exit(message=f"Buildenv workflow {conclusion}", code=1)
```

Comment:
This will become redundant if we move this check to the middle of the other function.